core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -292,11 +292,7 @@ void CudaInternal::initialize(cudaStream_t stream) {
   KOKKOS_IMPL_CUDA_SAFE_CALL(cudaError_t(cuCtxGetDevice(&m_cudaDev)));
   KOKKOS_IMPL_CUDA_SAFE_CALL(cudaSetDevice(m_cudaDev));
 
-  // FIXME_CUDA multiple devices
-  if (m_cudaDev != Cuda().cuda_device())
-    Kokkos::abort(
-        "Currently, the device id must match the device id used when Kokkos "
-        "was initialized!");
+  m_stream = stream;
```

Comment:
Not asking for a change, but moving this line from line 320 to here doesn't seem necessary.

Reply:
Yes, it is to use the correct stream in `scratch_unified`, `scratch_flags` and `scratch_space`.

Reply:
Ok I see it. https://github.com/kokkos/kokkos/blob/a1a6ea14c181b0e4f88cec82652bf594ceba5e22/core/src/Cuda/Kokkos_Cuda_Instance.cpp#L381-L385